config.toml

```diff
@@ -1,4 +1,72 @@
 [general]
 listen = "0.0.0.0:4242"
 
-[samplers]
+# The defaults are used for each sampler unless there's a sampler level
+# configuration present.
+
+[defaults]
+# Controls whether the samplers are enabled or not. Setting the default to
+# true means that individual sampler configs can be used to opt-out of
+# collection for that sampler. Setting the default to false requires that
+# individual sampler configs are used to opt-in to collection.
+enabled = true
+# The collection interval for counter and gauge based metrics. Shorter intervals
+# allow for more accurately capturing bursts in the related percentile metrics.
+interval = "10ms"
+# The collection interval for metrics that sample a distribution. Shorter
+# intervals reduce the uncertainty of the exact period corresponding to the
+# related percentile metrics.
+distribution_interval = "50ms"
+
+# Each sampler can then be individually configured to override the defaults. All
+# of the configuration options in the `[defaults]` section are allowed.
+
+# BPF sampler that instruments block_io request queue to measure latency and
+# size distribution.
+[samplers.block_io_latency]
+enabled = true
+
+# Produces various nVIDIA specific GPU metrics using NVML
+[samplers.gpu_nvidia]
+enabled = true
+
+# Memory utilization from /proc/meminfo
+[samplers.memory_meminfo]
+enabled = true
```

Comment:
Does this fail gracefully on machines without nvidia gpus? Or should be set it to disabled by default?

Reply:
Yes, this fails gracefully - an error is emitted during sampler init, but the sampler is effectively disabled at that point. It does not become part of the set of samplers and consumes no CPU time.